Workbooks/Azure Active Directory TroubleShooting/SensitiveOperations/SensitiveOperations.workbook

```diff
@@ -409,99 +406,73 @@
-                  "timeContext": {
-                    "durationMs": 86400000
-                  }
+                  "name": "ClientApp",
+                  "type": 1,
+                  "value": "All"
                 },
                 {
-                  "id": "86b52c65-055d-4a6a-a52f-b1d69e1f3c0f",
+                  "id": "2cadd1cb-53b7-41bb-b81f-e5dfcfdd4176",
                   "version": "KqlParameterItem/1.0",
-                  "name": "InitiatingUserOrApp",
+                  "name": "Resource",
                   "type": 1,
-                  "value": "All",
-                  "timeContext": {
-                    "durationMs": 86400000
-                  }
+                  "value": "All"
                 }
               ],
               "style": "pills",
               "queryType": 0,
               "resourceType": "microsoft.operationalinsights/workspaces"
             },
-            "name": "parameters - 1"
+            "name": "parameters - 4"
           },
           {
             "type": 3,
             "content": {
               "version": "KqlItem/1.0",
-              "query": "  AuditLogs\r\n  | where TimeGenerated {TimeRange:value}\r\n  | where \"{Operation}\" has OperationName\r\n  //| where Result =~ \"success\"   // commenting out, as it may be interesting to capture failed attempts\r\n  | extend targetDisplayName = tostring(TargetResources[0].displayName)\r\n  | extend UserAgent = iff(AdditionalDetails[0].key == \"User-Agent\",tostring(AdditionalDetails[0].value),\"\")\r\n  | extend InitiatingUserOrApp = iff(isnotempty(InitiatedBy.user.userPrincipalName),tostring(InitiatedBy.user.userPrincipalName), tostring(InitiatedBy.app.displayName))\r\n  | where InitiatingUserOrApp contains \"{InitiatingUserOrApp:escape}\" or \"{InitiatingUserOrApp:escape}\" == \"All\"\r\n  | project-reorder TimeGenerated, OperationName, InitiatingUserOrApp, AADOperationType, targetDisplayName, Result, UserAgent, CorrelationId, TenantId, AADTenantId\r\n  | sort by TimeGenerated desc",
+              "query": "AuditLogs\r\n| where TimeGenerated {TimeRange:value}\r\n| where OperationName == \"Add app role assignment to service principal\"\r\n| project TimeGenerated, ClientApp = tostring(TargetResources[0].modifiedProperties[4].newValue), Resource = tostring(TargetResources[0].displayName), Role_Added = tostring(TargetResources[0].modifiedProperties[1].newValue), Explanation = tostring(TargetResources[0].modifiedProperties[2].newValue), InitiatingUserOrApp = iff(isnotempty(InitiatedBy.user.userPrincipalName),tostring(InitiatedBy.user.userPrincipalName), tostring(InitiatedBy.app.displayName))\r\n| where ClientApp contains \"{ClientApp:escape}\" or \"{ClientApp:escape}\" == \"All\"\r\n| where Resource contains \"{Resource:escape}\" or \"{Resource:escape}\" == \"All\"\r\n| summarize by ClientApp, Resource, Role_Added, Explanation, InitiatingUserOrApp, TimeGenerated",
```

Comment:
is contains necessary here?

Reply:
Yes, because contains does a substring search on the name, whereas has looks for an exact match on the whole string.

Reply:
nit though:  when you use an expression multiple times, you might want to use `let` in the beginning like
```
let clientApp = "{ClientApp:escape}";
let resource = "{Resource:escape}";
```
then you can use those inline in the query, and they only get replaced once in the rest of the query text.   not a big deal for this specific one, but for list ones it can generate much smaller queries

Reply:
Thanks for the note. I have a lot of those all over the workbook but they're not searching through lists, so I'm not going to refactor the code now. In the future I will.